src/gui/qgsmapcanvas.cpp

```diff
@@ -147,6 +147,8 @@ QgsMapCanvas::QgsMapCanvas( QWidget *parent )
   mSettings.setFlag( QgsMapSettings::DrawEditingInfo );
   mSettings.setFlag( QgsMapSettings::UseRenderingOptimization );
   mSettings.setFlag( QgsMapSettings::RenderPartialOutput );
+  mSettings.setDestinationCrs( QgsProject::instance()->crs() );
+  mSettings.setEllipsoid( QgsProject::instance()->ellipsoid() );
```

Comment:
You'll also need to handle when the project ellipsoid is changed - listen out for the ellipsoidChanged signal and handle accordingly